CesiumGltf/src/MetadataFeatureTableView.cpp

```diff
@@ -124,9 +124,13 @@ MetadataPropertyViewStatus MetadataFeatureTableView::getBufferSafe(
     return MetadataPropertyViewStatus::InvalidValueBufferIndex;
   }
 
+  // TODO: investigate why this causes lots of metadata gltfs to fail, are they
+  // all incorrectly built?
```

Comment:
We should check with the 3D Tiles team on this, too, if you haven't already.

Reply:
Even at the risk of this being the result of confusion on my side (in terms of `EXT_feature_metadata` and `EXT_mesh_features`, and "Property Table" vs. "Feature Table"): The spec says at https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features#property-tables 

> > Each buffer view byteOffset must be aligned to a multiple of its component size.
>
> **Implementation note**: Authoring tools may choose to align all buffer views to 8-byte boundaries for consistency, but  client implementations should only depend on 8-byte alignment for buffer views containing 64-bit component types.

It **might** well be that some of the glTFs from https://github.com/CesiumGS/3d-tiles-samples/pull/40 are "invalid" in that way, because ... from a glTF perspective, a `bufferView` does not have "components". It's just raw data. The components are defined via the _accessor_, and here, we don't have an accessor ( c.f. https://github.com/CesiumGS/glTF/pull/44 ). 

From a quick glance, I think that the examples generally only have one buffer view with an offset of 0, so it would be good to know which examples fail exactly. 

Beyond that: The fact that a (core glTF) `bufferView` may have a "component type" _implicitly_ defined via the metadata that refers to it raises some questions. I'd consider to review this more thoroughly on a spec level,

Reply:
@javagl For EXT_feature_metadata (which is what this PR is aiming to support currently), the spec seems to say:
> Each buffer view byteOffset must be aligned to a multiple of 8 bytes. If the buffer view's buffer is the GLB-stored BIN chunk the byte offset is measured relative to the beginning of the GLB. Otherwise it is measured relative to the beginning of the buffer.

Reply:
So indeed seems to be something that changed in a subtle way between EXT_feature_metadata and EXT_mesh_features. I wonder how this implementation will eventually be "lifted up" to the EXT_mesh_features state. I've seen quite some `if`-statements for the "legacy" path in the CesiumJS code, wondered whether this will also be necessary here, and particularly, how such subtle changes like a changed/relaxed alignment requirement will be covered. But fingers crossed, many of the necessary changes might be derived from unit tests.